src/components/modals/EntityExtractor.tsx

```diff
@@ -303,16 +306,111 @@ class EntityExtractor extends React.Component<Props, ComponentState> {
         }
     }
 
+    static getUnique <T>(xs: T[], getKey: (x: T) => any): T[] {
+        const map = new Map<any, T>()
+
+        xs.forEach(x => {
+            const key = getKey(x)
+
+            if (!map.has(key)) {
+                map.set(key, x)
+            }
+        })
+
+        return Array.from(map.values())
+    }
+
+    /**
+     * Get all text variations that have same text but different entities
+     */
+    static getInconsistentResponses(trainDialogs: TrainDialog[], pendingExtractResponses: ExtractResponse[]): InconsistentExtractResponse[] {
+        // We assume all text variations within existing valid train dialogs are consistent
+        const textVarations = trainDialogs
+            .filter(td => !td.invalid)
```

Comment:
Check only happens at edit time, so seems like excluding invalid TDs could lead to inconsistent labeling.  If invalidated TD is fixed, no check will occur on the label to find the inconsistency